### ISSUE TITLE
Sync Path

### ISSUE THREAD

**Issue by anton-seaice** [payu-org/payu#462]
Should the sync path configure follow the style as short path? Where the experiment name is added to the path provided.

i.e. if:

```
sync: 
    enable: True
    path: /g/data/$PROJECT/$USER
```

should this create a folder 

`/g/data/$PROJECT/$USER/<<experiment_name+uuid>>`

automatically to sync to?

This seems especially useful when using the `payu branch` from a point in another experiment and you don't want to override the existing results ?

Current behavior is it will create the output folders directly in the _path_ directory

**Comment by jo-basevi** [comment]
Yeah that's a good idea. My only concern is that it would break existing usage of sync (if there is any), by syncing all output in a new subfolder in the `path` directory. But it would be good way to ensure uniqueness in remote archive folder name.

I thought I had previously added in logic in the sync code to check if there was pre-existing metadata.yaml file with a matching UUID in the sync path to ensure it was the same experiment, but after looking at the code, I had not sorry..

**Comment by anton-seaice** [comment]
I think its only an issue if users change payu version part way through an experiment? Or try to add to an existing experiment which was run on an older payu?

The worst case is that an extra folder is created - it wouldn't be too hard to tidy up manually ... ?

**Comment by jo-basevi** [comment]
Yeah true, it would not be difficult to tidy up manually, and would only be an issue if changing payu versions during running an experiment. 

Just a couple more thoughts about implementation:
- Experiments that were started prior to UUID changes (payu versions < 1.1), still have an experiment name (the name used for archive directories) being the control directory name. For these experiments, should the default archive folder name that is being synced to, include an UUID to ensure uniqueness?
- There could be an `experiment` option for the `sync` config, to allow overriding the default folder name if it's needed.

**Comment by anton-seaice** [comment]
@minghangli-uni - This would be a useful feature for the expt_manager tool. At the moment, its hard to use the sync feature with the pertubation tool, because the user would have to manually set the sync path for every perturbation. However, if payu was updated to create the `<<experiment_name+uuid>>` folder within `sync_path`, then the sync_path can be the same for every pertubation :)

**Comment by jo-basevi** [comment]
I think having a separate configuration option for  `base_path`/`base_dir`/? would be good. So `base_dir` would be the base directory and the payu experiment name is added on to it, and `path` would still the absolute override option - e.g. similar to `laboratory` vs `shortpath`. This will be easy to implement and maintains backwards compatibility.

**Comment by anton-seaice** [comment]
Is having two options for very similar functionality confusing?
What happen's if someone sets both?

**Comment by aidanheerdegen** [comment]
> 1. Someone has an experiment with sync enabled
> 2. They copy or clone that experiment, make some mods without changing the sync options and run it.
> 
> The output from 2 shouldn't override the output from 1.

With the current situation 2 does sync to the same location as 1.

The proposal for `base_dir` would fix this (assuming `experiment` is not set), because the top level directory is the same as the archive directory, which contains the unique hash for the experiment ID, and every new experiment should get a unique experiment ID.

**Comment by anton-seaice** [comment]
Sounds good to me :)

I think `path` could get deprecated in the future, with a warning from now.

But if we move all our released configs over to using `base_dir` (and set payu_min_version) then it's probably moot
